internal/observability/event/hclog_formatter_node_test.go

```diff
@@ -0,0 +1,368 @@
+			case true:
+				b, ok = e.Format(string(JSONHclogSinkFormat))
+			case false:
+				b, ok = e.Format(string(TextHclogSinkFormat))
+			}
+			t.Log(string(b))
+			require.True(ok)
+			for _, txt := range tt.want {
+				assert.Contains(string(b), txt)
+			}
+
+		})
+	}
+}
+
+func Test_hclogFormatterFilter_Name(t *testing.T) {
+	t.Parallel()
+	t.Run("simple", func(t *testing.T) {
+		ff := &hclogFormatterFilter{}
+		assert.Equal(t, hclogNodeName, ff.Name())
+	})
+}
+
+func Test_hclogFormatterFilter_Reopen(t *testing.T) {
+	t.Parallel()
+	t.Run("simple", func(t *testing.T) {
+		ff := &hclogFormatterFilter{}
+		assert.Equal(t, nil, ff.Reopen())
+	})
+}
+
+func Test_hclogFormatterFilter_Type(t *testing.T) {
+	t.Parallel()
+	t.Run("simple", func(t *testing.T) {
+		ff := &hclogFormatterFilter{}
+		assert.Equal(t, eventlogger.NodeTypeFormatterFilter, ff.Type())
+	})
+}
+
+func Test_newHclogFormatterFilter(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name            string
+		jsonFormat      bool
+		opt             []Option
+		wantErr         bool
+		wantIsError     error
+		wantErrContains string
+		wantAllow       []string
+		wantDeny        []string
+	}{
+		{
+			name: "no-opts",
+		},
+		{
+			name:       "bad-allow-filter",
+			jsonFormat: true,
+			opt: []Option{
+				WithAllow("foo=;22", "foo==bar"),
+			},
+			wantErr:         true,
+			wantErrContains: "invalid allow filter 'foo=;22'",
+		},
+		{
+			name:       "bad-deny-filter",
+			jsonFormat: true,
+			opt: []Option{
+				WithDeny("foo=;22", "foo==bar"),
+			},
+			wantErr:         true,
+			wantErrContains: "invalid deny filter 'foo=;22'",
+		},
+		{
+			name:       "empty-allow-filter",
+			jsonFormat: true,
+			opt: []Option{
+				WithAllow(""),
+			},
+			wantErr:         true,
+			wantErrContains: "missing filter",
+		},
+		{
+			name:       "empty-deny-filter",
+			jsonFormat: true,
+			opt: []Option{
+				WithDeny(""),
+			},
+			wantErr:         true,
+			wantErrContains: "missing filter",
+		},
+		{
+			name:       "valid-filters",
+			jsonFormat: true,
+			opt: []Option{
+				WithAllow("alice==friend", "bob==friend"),
+				WithDeny("eve==acquaintance", "fido!=dog"),
+			},
+			wantAllow: []string{"alice==friend", "bob==friend"},
+			wantDeny:  []string{"eve==acquaintance", "fido!=dog"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert, require := assert.New(t), require.New(t)
+			got, err := newHclogFormatterFilter(tt.jsonFormat, tt.opt...)
+			if tt.wantErr {
+				require.Error(err)
+				assert.Nil(got)
+				if tt.wantIsError != nil {
+					assert.ErrorIs(err, tt.wantIsError)
+				}
+				if tt.wantErrContains != "" {
+					assert.Contains(err.Error(), tt.wantErrContains)
+				}
+				return
+			}
+			require.NoError(err)
+			assert.NotNil(got)
+
+			assert.Equal(tt.jsonFormat, got.jsonFormat)
+
+			assert.Len(got.allow, len(tt.wantAllow))
+			for _, f := range got.allow {
+				assert.Contains(tt.wantAllow, f.raw)
```

Comment:
Should these be flipped (we are checking `f.raw` for something matching `tt.wantAllow`)?

Reply:
we're checking the list of all possible wants (a slice of strings) to see if a specific got string is present.